### PR TITLE
TIP-1192 Make the symfony kernel compatible with SF4

### PR DESCRIPTION
**Pull request by jjanvier**
For more information, please read http://fabien.potencier.org/symfony4-directory-structure.html and https://symfony.com/doc/current/bundles.html.

This is another step forward SF4 (and maybe Flex).

Next move is to remove those `AppKernelTest` classes :)

### CODE REVIEW

File: app/AppKernel.php [hunk 1]
```diff
@@ -12,194 +15,58 @@
...
     }
 
     /**
-     * @return string
+     * {@inheritdoc}
      */
     public function getLogDir(): string
     {
-        return dirname(__DIR__) . DIRECTORY_SEPARATOR . 'var' . DIRECTORY_SEPARATOR . 'logs';
+        return $this->getProjectDir().'/var/logs';
```
Comment (ahocquard) [inline app/AppKernel.php]: `DIRECTORY_SEPARATOR`?

Comment (anaelChardan) [reply app/AppKernel.php]: Don’t know if it is useful as we only target linux.

File: app/AppKernel.php [hunk 2]
```diff
@@ -12,194 +15,58 @@
...
-            new Acme\Bundle\AppBundle\AcmeAppBundle(),
-        ];
-    }
+    use MicroKernelTrait;
 
     /**
      * {@inheritdoc}
      */
-    public function registerBundles()
+    public function registerBundles(): iterable
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply app/AppKernel.php]: you should do that in EE too :)

File: app/AppKernel.php (hunk 1 above)
Comment (arnolanglade) [reply app/AppKernel.php]: Symfony does not use them neither!

File: tests/back/Integration/AppKernelTest.php [hunk 3]
```diff
@@ -12,41 +10,14 @@
...
     /**
-     * {@inheritdoc}
+     * {@inheritDoc}
+     *
+     * Necessary to make gedmo extension tree work. Otherwise the path located in
+     * "vendor/akeneo/pim-community-dev/src/Akeneo/Platform/config/bundles/gedmo_doctrine_extensions.yml"
+     * is never the right one...
      */
-    protected function registerProjectBundles(): array
+    public function getRootDir()
```
Comment (christophe-chausseray) [inline tests/back/Integration/AppKernelTest.php]:
```suggestion
    public function getRootDir(): string
```

**Pull request merged**